src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.Max.cs

```diff
@@ -74,73 +75,64 @@ public static void Max<T>(ReadOnlySpan<T> x, ReadOnlySpan<T> y, Span<T> destinat
         /// </remarks>
         public static void Max<T>(ReadOnlySpan<T> x, T y, Span<T> destination)
             where T : INumber<T> =>
-            InvokeSpanScalarIntoSpan<T, MaxPropagateNaNOperator<T>>(x, y, destination);
+            InvokeSpanScalarIntoSpan<T, MaxOperator<T>>(x, y, destination);
 
-        /// <summary>T.Max(x, y) (but NaNs may not be propagated)</summary>
-        internal readonly struct MaxOperator<T> : IAggregationOperator<T> where T : INumber<T>
+        /// <summary>Max(x, y)</summary>
+        internal readonly struct MaxOperator<T> : IAggregationOperator<T>
+             where T : INumber<T>
         {
             public static bool Vectorizable => true;
 
-            public static T Invoke(T x, T y)
-            {
-                if (typeof(T) == typeof(Half) || typeof(T) == typeof(float) || typeof(T) == typeof(double))
-                {
-                    return x == y ?
-                        (IsNegative(x) ? y : x) :
-                        (y > x ? y : x);
-                }
-
-                return T.Max(x, y);
-            }
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static T Invoke(T x, T y) => T.Max(x, y);
 
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             public static Vector128<T> Invoke(Vector128<T> x, Vector128<T> y)
             {
-                if (typeof(T) == typeof(float) || typeof(T) == typeof(double))
+#if !NET9_0_OR_GREATER
```

Comment:
What informed the removal of this specialization in .NET 9?

Reply:
It's not needed and `Vector128.Max` does the correct thing on .NET 9+